_posts/2021-08-20-gsoc-project-summary.md

```diff
@@ -0,0 +1,55 @@
+---
+layout: post
+title: "GSoC World-Gen Project: Wrap Up" #This is your post tile
+description: > #This is a brief description of your post that will show up in post previews.
+    The last few weeks have brought the project to completion! We've now got fully functioning and modular World Generation in Destination: Sol
+author: "IsaiahBlanks1" #This will show up as the post author
+image: "gooey-imperial.png" #This is the image that you see in post previews
+parallax: true #This determines whether or not the asteroid field will appear
+---
+Summary
+-------
+    
+It's time to wrap up the 2021 GSoC project for Destination: Sol. Over the last couple of months, I've entirely redesigned the world generation system in Destination: Sol. Previously, the Destination: Sol world always consisted of the same few Planets, Mazes and, Asteroid Belts, all within the same type of SolarSystem. Adding any new type of feature to the game would have been a hassle, as every aspect of the world-gen system was tightly coupled to every other aspect. This made it hard to change any particular part of the world.
+
+Now the world generation process works quite differently. It begins with `GalaxyBuilder` (previously referred to as `WorldBuilder`). This class uses reflection to find all available `SolarSystemGenerator` classes. A `Generator` class represents a feature of the Destination: Sol world, such as a Planet, Maze, Asteroid Belt, or even a SolarSystem. The `GalaxyBuilder` loads `Generators` that represent entire SolarSystems. It then finds a position for each SolarSystem (based on the SolarSystem size) and then tells each `SolarSystemGenerator` to run its build process.
+
+This build process has two sections. First, properties related to this specific `SolarSystemGenerator`, such as its name or its config file, are set. Then, a method is called to initiate the build process of each Feature (Planet, Maze, or Asteroid Belt) that will populate this SolarSystem. Then, once each of those build processes complete, the `SolarSystemGenerator` creates a `SolarSystem` object for the game's runtime code to use. 
+
+We mentioned that each of those `SolarSystemGenerator` build processes initiates `FeatureGenerator` build processes. These work very similarly to the process for `SolarSystemGenerators`. However, they only contain the step of setting the details specific to that feature. For example, within the `build()` method of a `PlanetGenerator` implementation, the code will set the Planet config file, ground height, atmosphere height, orbit speed, rotation speed, and name. Once the values are set, the `build()` method concludes by instantiating a `Planet` object for the game to use.
```

Comment:
Trivial: "We mentioned" to me sounds weird in this context, it sounds _conversational_ when a blog entry or wiki article tends to be from a more passive point of view. Although I might be slipping that personal preference into mediums where the conversational style _can_ work (places like javadoc are harder to justify as conversational, for instance) - so this is really more of a comment than a suggestion to change.